objects/src/accounts/storage/mod.rs

```diff
@@ -299,7 +348,28 @@ impl Deserializable for AccountStorage {
             });
         }
 
-        Self::new(items).map_err(|err| DeserializationError::InvalidValue(err.to_string()))
+        // read the number of StorageMap instances
+        let num_storage_maps = source.read_u8()?;
+
+        let mut maps = Vec::with_capacity(num_storage_maps as usize);
+
+        for _ in 0..num_storage_maps {
+            // read the number of filled leaves for this StorageMap
+            let num_filled_leaves = source.read_u8()?;
+            let mut entries = Vec::with_capacity(num_filled_leaves as usize);
+
+            for _ in 0..num_filled_leaves {
+                let key = source.read()?;
+                let value = source.read()?;
+                entries.push((key, value));
+            }
+            maps.push(
+                StorageMap::with_entries(entries)
+                    .map_err(|err| DeserializationError::InvalidValue(err.to_string()))?,
+            );
+        }
+
+        Self::new(items, maps).map_err(|err| DeserializationError::InvalidValue(err.to_string()))
```

Comment:
**Request**: The changes to the deserializer should match changes mentioned for the serializer requested above.